src/delete_backups.py

```diff
@@ -0,0 +1,218 @@
+"""
+Delete backups in S3 bucket according to our Data Rention Policy:
+https://github.com/aspiredu/aspiredu/issues/7421
+"""
+import argparse
+import os
+import re
+from collections import defaultdict
+from datetime import date, datetime, timedelta
+from functools import lru_cache
+from itertools import chain
+
+import sentry_sdk
+from dateutil import rrule
+from dateutil.relativedelta import SA, relativedelta
+from dotenv import load_dotenv
+
+from src.s3 import get_s3
+
+# ENV Variables
+load_dotenv()
+
+ASPIRE_AWS_ACCESS_KEY_ID = os.getenv("ASPIRE_AWS_ACCESS_KEY_ID")
+ASPIRE_AWS_SECRET_ACCESS_KEY = os.getenv("ASPIRE_AWS_SECRET_ACCESS_KEY")
+SENTRY_DSN = os.getenv("SENTRY_DSN")
+
+CRUNCHYBRIDGE_BACKUP_PATTERN = re.compile(r"(\d{8})-\w*")
+
+
+class TooManyDirectoriesForDeletion(Exception):
+    pass
+
+
+def is_saturday(value: date) -> bool:
+    """Determine if date is a Saturday"""
+    return value.weekday() == 5
+
+
+@lru_cache(maxsize=None)
+def saturdays_for_the_past_year() -> set[date]:
+    """Find all Saturdays between from today to 1 year ago"""
+    start = date.today()
+    cut_off = start - relativedelta(years=1)
+    return set(
+        d.date() for d in rrule.rrule(rrule.WEEKLY, byweekday=SA(1), dtstart=cut_off, until=start)
+    )
+
+
+@lru_cache(maxsize=None)
+def first_saturdays_of_month_year1_to_year3() -> set[date]:
+    """Find first Saturday of each month between from 1 year to 3 years"""
+    today = date.today()
+    start = today - relativedelta(years=1)
+    cut_off = today - relativedelta(years=3)
+    return set(
+        d.date() for d in rrule.rrule(rrule.MONTHLY, byweekday=SA(1), dtstart=cut_off, until=start)
+    )
+
+
+def meets_retention_policy(value: date):
+    today = date.today()
+    if (today - timedelta(weeks=2)) < value <= today:
```

Comment:
A comment here to point out that the retention policy of retaining all back ups for 2 weeks is being enforced here should make it easier for someone in the future to adjust as necessary.